python/tests/dynamic_binding_test.py

```diff
@@ -67,7 +67,11 @@ def test_bound_methods_groupby():
     _, da = make_containers()
     da.coords['x'] = sc.array(dims=['x'],
                               values=rng.choice([0, 1], da.coords['x'].shape))
-    assert sc.identical(da.groupby('x').sum('x'), sc.groupby(da, 'x').sum('x'))
+    assert sc.identical(
+        da.groupby(group='x').sum('x'),
+        sc.groupby(da, group='x').sum('x'))
```

Comment:
> `da.groupby('x')` should be obvious?

Not sure I agree. Since `groupby` has this additional function of binning, it is not super clear which argument is which unless you name them.

But the main problem is, can the generic binding function decide which arguments should be keyword only and which positional? It would be hard to take subtleties like arguments with the same type into account. But of course we can bind `groupby` manually.

Reply:
But isn't the first (group) arg always required, even if bins are given?

Reply:
Yes. But that was never part of the discussion on requiring keywords. E.g. `array` requires `dims` and `values` but they are still keyword-only.

Reply:
You are right that it is unclear. Personally I feel that any function that has exactly 1 required argument is ok without keywords. For example:
- `sc.scalar` works without `value=`.
- `sc.sum(da)` works
- `da.groupby('x')` feels natural

Isn't one of the main reasons for keyword args to avoid mixing up arg order? In none of the above cases there is any risk for that, I believe?

It gets more blurry for things such as `sc.sum(da, 'x')`, which has 2 args. I still feel that is fine, but maybe that is just because I am used to it? Maybe because `da.sum('x')` does the same and is also clear?

Reply:
How about this guideline (exceptions may be considered individually):
- Functions with a single arg are ok (`sc.sin(da)`).
- Functions with two args are ok provided that:
  - Both args are scipp objects (`sc.less(da1, da2)`). Exception: `sc.atan2` which has a potentially confusing arg order
  - Second args is a dim (`sc.sum(var, 'x')`)
- Any further optional args must be keyword args.
- Any functions that are not in the categories above should generally require keyword args.

Reply:
I mostly agree. But what about `to_unit`?
Maybe we can change ii. to functions where the first arg is a scipp object. Those can be seen analogous to methods in a Universal Function Call Syntax kind of way. In this case it would always be ok to have the first and potential second arg positional. All additional args would be keyword-only.

Reply:
Suggestion above sounds reasonable to me. Seems to balance enforcing keyword arguments where we really should avoid ambiguity and unnecessary typing where there is none. The `atan2` example is maybe one to bear in mind as we implement new things, I can't think of any strong similar examples at present (maybe cross product if we had that), but there may be other exceptions like this.

Reply:
I also agree that you should allow non-keyword arguments where you only have one argument, or when using free functions like `sc.sum()`.
I think users would be rapidly annoyed if they have to use kwargs everywhere.